src/Application/Engine.cpp

```
@@ -258,12 +259,13 @@ void Engine::startThreadMonitor() {
-                if (g_appCtx.MainThread.isHalted.load()) {
-                    m_eventDispatcher->dispatch(UpdateEvent::ApplicationStatus{
-                        .appState = Application::State::IDLE
-                    }, false, true);
+                if (g_appCtx.MainThread.isHalted.load(std::memory_order_relaxed)) {
+                    //m_eventDispatcher->dispatch(UpdateEvent::ApplicationStatus{
+                    //    .appState = Application::State::IDLE
+                    //}, false, true);
+                    //ThreadManager::SignalMainThreadResume();
 
-                    ThreadManager::SignalMainThreadResume();
+                    g_appCtx.MainThread.isHalted.store(false, std::memory_order_relaxed);
```

Cppcheck (2.20.0)

[performance] Variable 'm_requiredMask' is assigned in constructor body. Consider performing initialization in initialization list.

(useInitializationList)